.github/workflows/ci.yml

```diff
@@ -0,0 +1,265 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - dev
+      - stable
+    paths-ignore:
+      - 'documentation/**'
+      - '**.md'
+  pull_request:
+    branches:
+      - dev
+      - stable
+    paths-ignore:
+      - 'documentation/**'
+      - '**.md'
+
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: ${{ matrix.os }}
```

Comment:
It triggers me a tiny bit that we use matrix before defining it :trollface:

Reply:
Ah the beauty of declarative languages :)